src/libs/actions/Report.ts

```diff
@@ -2930,6 +2947,215 @@ function getReportPrivateNote(reportID: string | undefined) {
+        const instructionMessage: TaskMessage = {
+            reportID: currentTask.reportID,
+            reportActionID: instructionCommentAction.reportActionID,
+            reportComment: instructionCommentText,
+        };
+
+        return [
+            ...acc,
+            {
+                type: 'task',
+                task: engagementChoice,
+                taskReportID: currentTask.reportID,
+                parentReportID: currentTask.parentReportID ?? '',
+                parentReportActionID: taskReportAction.reportAction.reportActionID,
+                assigneeChatReportID: '',
+                createdTaskReportActionID: taskCreatedAction.reportActionID,
+                title: currentTask.reportName ?? '',
+                description: currentTask.description ?? '',
+            },
+            {
+                type: 'message',
+                ...subtitleMessage,
+            },
+            {
+                type: 'message',
+                ...instructionMessage,
+            },
+        ];
+    }, []);
+
+    const tasksForOptimisticData = tasksData.reduce<OnyxUpdate[]>((acc, {currentTask, taskCreatedAction, taskReportAction, subtitleComment, instructionComment}) => {
+        const subtitleCommentAction: OptimisticAddCommentReportAction = subtitleComment.reportAction;
+        const instructionCommentAction: OptimisticAddCommentReportAction = instructionComment.reportAction;
+
+        return [
+            ...acc,
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${targetChatReportID}`,
+                value: {
+                    [taskReportAction.reportAction.reportActionID]: taskReportAction.reportAction as ReportAction,
+                },
+            },
+            {
+                onyxMethod: Onyx.METHOD.SET,
+                key: `${ONYXKEYS.COLLECTION.REPORT}${currentTask.reportID}`,
+                value: {
+                    ...currentTask,
+                    pendingFields: {
+                        createChat: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                        reportName: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                        description: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                        managerID: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                    },
+                    isOptimisticReport: true,
+                },
+            },
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${currentTask.reportID}`,
+                value: {
+                    [taskCreatedAction.reportActionID]: taskCreatedAction as ReportAction,
+                    [subtitleCommentAction.reportActionID]: subtitleCommentAction as ReportAction,
+                    [instructionCommentAction.reportActionID]: instructionCommentAction as ReportAction,
+                },
+            },
+        ];
+    }, []);
+
```

Comment:
It already uses the `OnyxUpdate[]` type. This data is not being sent to the backend, because it is the optimistic data. So the app checks if we use proper optimistic data.